presto-main/src/main/java/com/facebook/presto/operator/WindowInfo.java

```diff
@@ -0,0 +1,252 @@
+            this.sumSquaredDifferencesSizeOfIndex = sumSquaredDifferencesSizeOfIndex;
+            this.sumSquaredDifferencesSizeInPartition = sumSquaredDifferencesSizeInPartition;
+            this.totalPartitionsCount = totalPartitionsCount;
+            this.totalRowsCount = totalRowsCount;
+            this.numberOfIndexes = numberOfIndexes;
+        }
+
+        @JsonProperty
+        public double getSumSquaredDifferencesPositionsOfIndex()
+        {
+            return sumSquaredDifferencesPositionsOfIndex;
+        }
+
+        @JsonProperty
+        public double getSumSquaredDifferencesSizeOfIndex()
+        {
+            return sumSquaredDifferencesSizeOfIndex;
+        }
+
+        @JsonProperty
+        public double getSumSquaredDifferencesSizeInPartition()
+        {
+            return sumSquaredDifferencesSizeInPartition;
+        }
+
+        @JsonProperty
+        public long getTotalPartitionsCount()
+        {
+            return totalPartitionsCount;
+        }
+
+        @JsonProperty
+        public long getTotalRowsCount()
+        {
+            return totalRowsCount;
+        }
+
+        @JsonProperty
+        public long getNumberOfIndexes()
+        {
+            return numberOfIndexes;
+        }
+    }
+
+    @ThreadSafe
+    private static class IndexInfoBuilder
+    {
+        private final long rowsNumber;
+        private final long sizeInBytes;
+        private final ImmutableList.Builder<Integer> partitionsSizes = ImmutableList.builder();
+
+        public IndexInfoBuilder(long rowsNumber, long sizeInBytes)
+        {
+            this.rowsNumber = rowsNumber;
+            this.sizeInBytes = sizeInBytes;
+        }
+
+        public synchronized void addPartition(WindowPartition partition)
+        {
+            partitionsSizes.add(partition.getPartitionEnd() - partition.getPartitionStart());
+        }
+
+        public synchronized Optional<IndexInfo> build()
+        {
+            List<Integer> partitions = partitionsSizes.build();
+            if (partitions.size() == 0) {
+                return Optional.empty();
+            }
+            double avgSize = partitions.stream().mapToLong(Integer::longValue).average().orElse(Double.NaN);
+            double squaredDifferences = partitions.stream().mapToDouble(size -> Math.pow(size - avgSize, 2)).sum();
+            checkState(partitions.stream().mapToLong(Integer::longValue).sum() == rowsNumber, "Total number of rows in index does not match number of rows in partitions within that index");
+
+            return Optional.of(new IndexInfo(rowsNumber, sizeInBytes, squaredDifferences, partitions.size()));
+        }
+    }
+
+    @Immutable
+    private static class IndexInfo
+    {
+        private final long totalRowsCount;
+        private final long sizeInBytes;
+        private final double sumSquaredDifferences; // sum of (partitionSize - averagePartitionSize)^2 for each partition
```

Comment:
make more meaningful name as in: https://github.com/prestodb/presto/pull/7661/commits/afcacf6ddb515141c43c6b2ff7f209f9cee9bfda#diff-44e86b0fa3217812de1a11e39ecfeaabR123